apm-protocol/apm-network/src/main/java/org/apache/skywalking/apm/network/trace/component/ComponentsDefine.java

```diff
@@ -229,4 +229,6 @@ public class ComponentsDefine {
 
     public static final OfficialComponent NATS = new OfficialComponent(132, "Nats");
 
+    public static final OfficialComponent IMPALA_JDBC_DRIVER = new OfficialComponent(133, "Impala-jdbc-driver");
```

Comment:
The component ID should be registered in the main repo first.
And as this component ID is a client driver, you should register 2 IDs, one for this client and the other for impala server. Also, you need ID mapping in the same config file.

Ref to https://skywalking.apache.org/docs/skywalking-java/latest/en/setup/service-agent/java-agent/java-plugin-development-guide/#abstractspan

> Component IDs are defined and reserved by the SkyWalking project. For extension of the component name/ID, please follow the OAP server Component library settings document.

Logo could be submitted to the UI repo.

Reply:
Okey.